pkg/trace/writer/trace.go

```diff
@@ -34,14 +35,14 @@ var MaxPayloadSize = 3200000 // 3.2MB is the maximum allowed by the Datadog API
 
 // SampledSpans represents the result of a trace sampling operation.
 type SampledSpans struct {
-	// Trace will contain a trace if it was sampled or be empty if it wasn't.
-	Traces []*pb.APITrace
-	// Events contains all APM events extracted from a trace. If no events were extracted, it will be empty.
-	Events []*pb.Span
+	// TracerPayload will contain trace chunks if it was sampled or be empty if it wasn't.
+	TracerPayload *pb.TracerPayload
 	// Size represents the approximated message size in bytes.
 	Size int
 	// SpanCount specifies the total number of spans found in Traces.
```

Comment:
We need to update the documentation here, due to https://github.com/DataDog/datadog-agent/pull/9474/files#r739153296:
```suggestion
	// SpanCount specifies the number of spans that were sampled as part of a trace inside the TracerPayload.
	// It excludes spans sampled as events.
```
This will help resolve the other comment.